### ISSUE TITLE
Tiny corner bug (kcorner bug?) on the bottom edges

### ISSUE THREAD

**Issue by levy-deng** [matinlotfali/KDE-Rounded-Corners#100]
When the blur effect is enabled, there is a tiny corner bug on the bottom of the screen. I'm unsure why that's the case, but it seems like the kcorners bug from all the way back.

**Comment by matinlotfali** [comment]
I confirm that I sometimes see that on a terminal with blurred background. 
I don't know how to fix this. I will be needing some assistance to figure it out.

**Comment by levy-deng** [comment]
let me look into how the kcorners bug fix worked, but from what I've read from various sources and an educated guess, it is due to the blur affect still interpreting the window size prior to applying the shader.

**Comment by levy-deng** [comment]
One detail to note: There is no bug if you use this shader with the breeze theme.

**Comment by matinlotfali** [comment]
Can you post a screenshot?

**Comment by RomanKode** [comment]
Apparently the "Korner Bug" fix was either removed of broken in new Plasma versions, there was a merge request for an API to add the feature and that feature would had also addressed the lack of blur on morphing windows(Magic lamp and Woobly) but in the end nothing happened, I would suggest to bring that up to the KDE devs and see if there could be a possible solution. So far the only solution I can see is if it's implemented at the Application Style level(Fork breeze, Lightly or even Klassy and implement it there. The Korner bug "fixed" at the Decoration Style level but not at the Application Style level.

**Comment by RomanKode** [comment]
I know it would add complexities to the code, but probably this can help. What if it were possible to become the blur plugin and control its radius from the same plugin that controls the borders?

https://github.com/Alban-Boissard/kwin-effects-blur-respect-rounded-decorations

**Comment by levy-deng** [comment]
One more thing to note: if selected no borders on a custom theme the corners go away. I think this might have something to do with the blur shader and borders somehow?

**Comment by matinlotfali** [comment]
@levy-deng I am applying the rounding shader on all windows with plasma decoration. If a window comes with no decoration, you can add the window to the inclusion list in settings. The feature is added recently in the `master` branch.

**Comment by matinlotfali** [comment]
Is this issue resolved in the latest releases?
@levy-deng @RomanKode

**Comment by levy-deng** [comment]
Yep! Issue is resolved. need to have the border rule set to no borders.On Jan 22, 2024, at 9:54 AM, Matin Lotfaliei ***@***.***> wrote:﻿
Is this issue resolved in the latest releases?
@levy-deng @RomanKode

**Comment by RomanKode** [comment]
> Is this issue resolved in the latest releases? @levy-deng @RomanKode

No and it may never will, unless KDE Devs add a Kwin API to do so.

**Comment by RomanKode** [comment]
![Screenshot_20240122_171933](https://github.com/matinlotfali/KDE-Rounded-Corners/assets/70252656/4b40b7c1-43da-468c-8986-690780cfdde1)
Still has the Korner Bug, just that shadows hide it and this a recently compiled version of this effect.

This is what the dev of Lightlyshaders is doing for his Plasma 6 release of the effect.
https://github.com/a-parhom/LightlyShaders#plasma-6
Basically forking his own Blur effect to be included.

**Comment by queenbiscuit311** [comment]
just thought i should heads up, on plasma 6 if you install the lightlyshaders plasma 6 branch, use the lightly shaders blur, disable the normal blur, and then enable this extension the korners bug is fixed. at least it would appear to be.

**Comment by matinlotfali** [comment]
@queenbiscuit311 It should be similar in KDE-Rounded-Corners

**Comment by KainNhantumbo** [comment]
I just wanted to point that this corner bug appears on blurred windows like terminal in plasma 6, before applying this effect.

**Comment by RomanKode** [comment]
Checkout Queenbiscuit's comments, it works if you use lightlyshaders blur plugin

**Comment by thrdroom** [comment]
> ![Screenshot_20240122_171933](https://private-user-images.githubusercontent.com/70252656/298755604-4b40b7c1-43da-468c-8986-690780cfdde1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjkxODUzOTYsIm5iZiI6MTcyOTE4NTA5NiwicGF0aCI6Ii83MDI1MjY1Ni8yOTg3NTU2MDQtNGI0MGI3YzEtNDNkYS00NjhjLTg5ODYtNjkwNzgwY2ZkZGUxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDE3VDE3MTEzNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVhYTcwMWFkZGFhNjEwZjM1ODEwNmFjNDk0NzE0OTZhODMxNmY1ZDI2NTNiMmVmNmU5OTE2NzY2NGRmZTJlNjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ttQ7s5vSQdKIcgjk21L85Z0C19O0ljVD_zFq5y2GwZY) Still has the Korner Bug, just that shadows hide it and this a recently compiled version of this effect.
> 
> This is what the dev of Lightlyshaders is doing for his Plasma 6 release of the effect. https://github.com/a-parhom/LightlyShaders#plasma-6 Basically forking his own Blur effect to be included.

Can someone **please** bring this fix to the official kwin devs and **force them** to finally fix their blur effect so we can finally get rid of the corner bug? Who can we ping here from KDE/kwin to look into this?

**Comment by omuas** [comment]
This blur effect has options for rounded corners.
https://github.com/taj-ny/kwin-effects-forceblur
